### DISCUSSION TITLE
My Elastalert rule won't work

### DISCUSSION THREAD

**Discussion by nekomimi-mode** [Security-Onion-Solutions/securityonion#11244] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

10

### RAM

38GiB

### Storage for /

237GiB

### Storage for /nsm

1.9TiB

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

Hi,

I copied my Elastalert rule which is working on 2.3.260 to 2.4.10, but it won't work well.

I'm running both 2.3.260 and 2.4.10, and both gets same alerts from Suricata.
When getting an alert, 2.3.260 works correctly, but logs of 2.4.10 always say "`0 query hits (0 already seen), 0 matches, 0 alerts sent`."
A major difference during install is choosing IP (2.3.260) and FQDN (2.4.10) for Web Access method.

Please help me.

Here's the rule saved as `/opt/so/rules/elastalert/suricata.yaml`.
```
# Elastalert rule to forward Suricata alerts from Security Onion to Mail.

name: Suricata-Alert
type: any
index: "*:so-ids-*"
buffer_time:
    minutes: 5
query_key: ["event.severity_label","rule.uuid","rule.name","source.ip","destination.ip"]
realert:
    minutes: 5
filter:
- query:
   query_string:
      query: "event.module: suricata AND rule.severity: (1 OR 2)"

alert:
- "email"
email: "root"
smtp_host: "<IP ADDRESS of SMTP>"
smtp_port: 25
from_addr: "oel9soc@<MY FQDN>"

alert_subject: "{5}:{6} {1}:{2} -> {3}:{4} {0}"
alert_subject_args: ["@timestamp","source.ip","source.port","destination.ip","destination.port","rule.uuid","rule.name"]

alert_text_type: alert_text_only
alert_text: "SID        : {7}\n
Rule Nmae  : {0}\n
Severity   : {1}\n
Time       : {2}\n
Sorce      : {3}:{4}\n
Destination: {5}:{6}\n
=====Payload==========\n
{8}"

alert_text_args: ["rule.name","rule.severity","@timestamp","source.ip","source.port","destination.ip","destination.port","rule.uuid","network.data.decoded"]
```
It seems syntax is correct because no error in `/opt/so/log/elastalert/stderr.log` after running `so-elastalert-restart`.
Following logs are shown every 3 minutes in `/opt/so/log/elastalert/elastalert.log`:
```
2023-09-02 05:59:59,367     INFO apscheduler.executors.default Running job "Rule: Suricata-Alert (trigger: interval[0:03:00], next run at: 2023-09-02 06:03:03 UTC)" (scheduled at 2023-09-02 05:59:59.366766+00:00)
2023-09-02 05:59:59,370     INFO           elastalert Queried rule Suricata-Alert from 2023-09-02 05:55 UTC to 2023-09-02 05:59 UTC: 0 / 0 hits
2023-09-02 05:59:59,372  WARNING        elasticsearch DELETE https://oel9soc:9200/_search/scroll [status:404 request:0.001s]
2023-09-02 05:59:59,375     INFO           elastalert Ran Suricata-Alert from 2023-09-02 05:55 UTC to 2023-09-02 05:59 UTC: 0 query hits (0 already seen), 0 matches, 0 alerts sent
2023-09-02 05:59:59,376     INFO           elastalert Suricata-Alert range 244
2023-09-02 05:59:59,376     INFO apscheduler.executors.default Job "Rule: Suricata-Alert (trigger: interval[0:03:00], next run at: 2023-09-02 06:03:03 UTC)" executed successfully
```

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by dougburks** [comment, marked as answer]
There are many changes from 2.3 to 2.4, so you'll probably need to update the index value (and perhaps others):
https://docs.securityonion.net/en/2.4/elasticsearch.html?#indexing

**Reply by nekomimi-mode** [reply]
Thank you, Doug!
I got it!

I've changed the index value to `index: "*:*logs-suricata-*"`, then it works!
